tests/acceptance/30_custom_promise_types/05_cp_using_library.cf

```diff
@@ -0,0 +1,74 @@
+######################################################
+#
+#  Basic test of promise module using the shell library
+#
+#####################################################
+body common control
+{
+    inputs => { "../default.cf.sub" };
+    bundlesequence  => { default("$(this.promise_filename)") };
+    version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+  vars:
+    "test_string"
+      string => "hello, modules";
+
+   files:
+    "$(G.testfile).src"
+      create => "true",
+      edit_line => insert_lines("$(test_string)");
+
+    "$(G.testfile).dst"
+      delete => init_delete;
+
+    "$(this.promise_dirname)/cfengine.sh"
+      copy_from => local_cp("$(this.promise_dirname)/../../../docs/custom_promise_types/cfengine.sh");
+
+}
+
+body delete init_delete
+{
+      dirlinks => "delete";
+      rmdirs   => "true";
+}
+
+#######################################################
+
+promise agent example
+{
+    interpreter => "/bin/bash";
```

Comment:
Is this going to work on all exotics?

Reply:
well, the test passes :) So I believe it means the file `/bin/bash` exists on all exotics, and is capable of parsing shell scripts.